Modix/Modules/UserInfoModule.cs

```diff
@@ -0,0 +1,60 @@
+﻿namespace Modix.Modules
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Discord;
+    using Discord.Commands;
+    using Discord.WebSocket;
+    using Humanizer;
+    using Humanizer.Localisation;
+
+    [Summary("Used to show information about a user.")]
+    public class UserInfoModule : ModuleBase
+    {
+
+        [Command("info"), Summary("Shows Information for a specific user")]
+        public async Task Info(SocketUser user = null)
+        {
+            var userInfo = user ?? Context.User;
+            var uname = userInfo.Username ?? "?";
+
+            var builder = new EmbedBuilder
+            {
+                Author = new EmbedAuthorBuilder
+                {
+                    Name = userInfo.IsBot ? $"{uname} (Bot)" : $"{uname}",
+                    IconUrl = userInfo.GetAvatarUrl()
+                },
+                Color = new Color(24, 138, 184),
+                ThumbnailUrl = userInfo.GetAvatarUrl()
+            };
+            var socketUser = userInfo as SocketGuildUser;
+            builder.WithTitle("User Information");
+;           builder.AddInlineField("ID: ", userInfo.Id);
+            if (socketUser.Nickname != null) builder.AddInlineField("Nickname: ", socketUser.Nickname);
+            builder.AddInlineField("Profile: ", userInfo.Mention);
+            builder.AddInlineField("Created: ", $"{(DateTime.Now - userInfo.CreatedAt).Humanize(2, null, TimeUnit.Year)} ago ({userInfo.CreatedAt})");
+            builder.AddInlineField("Current Status: ", userInfo.Status);
+            builder.AddInlineField("Joined: ", $"{(DateTimeOffset.Now - socketUser.JoinedAt.Value).Humanize(2, null, TimeUnit.Year)} ago ({socketUser.JoinedAt})");
+
+            var channels = await Context.Guild.GetTextChannelsAsync();
+
+            var messages = new List<IMessage>();
+            foreach (var channel in channels)
+            {
+                messages.AddRange(channel.GetMessagesAsync().Flatten().Result.ToList());
+            }
+
```

Comment:
How does this behave for someone who has been on the server for a long time and has many, many, many messages? (me?)

My thought is that this generates a lot of api calls, which will result in the bot getting throttled; or the call is slightly smarter and only returns the result of 1 api call. in which case this value will be inaccurate. Rowboat does this by storing the value in their own database.

Reply:
Right now this would be multiple api calls as it  has to get all text channels and then subsequently get all messages from those channels. The idea was to refactor this out and use the database instead but I wasn't sure if it was okay and add all messages as this could be quite large.

Reply:
Gonna refactor this tonight to make it query the database instead of doing a .GetMesssgesAsync(). Will leave the pull requests open and push to it later. :)